fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java

```diff
@@ -3689,9 +3700,11 @@ private Money calculateTotalOverpayment() {
         }
 
         for (final LoanTransaction loanTransaction : this.loanTransactions) {
-            if ((loanTransaction.isRefund() || loanTransaction.isRefundForActiveLoan() || loanTransaction.isCreditBalanceRefund())
-                    && !loanTransaction.isReversed()) {
-                totalPaidInRepayments = totalPaidInRepayments.minus(loanTransaction.getAmount(currency));
+            if (!loanTransaction.isReversed()) {
+                if ((loanTransaction.isRefund() || loanTransaction.isRefundForActiveLoan() || loanTransaction.isCreditBalanceRefund()
+                        || loanTransaction.isChargeback())) {
```

Comment:
I dont think this is right... Chargeback is kind of the opposite of refund. In my understanding when the customer gets refund it is like an opposite-way repayment, if the customer has $50 unpaid anything and in the event of $20 refund, the remaining balance will be $30. However with chargeback we are not repaying any credit, we are increasing that.

@josehernandezfintecheandomx What am I missing here?

Reply:
This is for `calculateTotalOverpayment` so, for that you get all the repayments, then all the amounts paid in the different installments

In this part that you comment is to consider all the amounts that were returned for X reason, refunds in general and charge back too